tests/src/main/java/com/hazelcast/stabilizer/tests/helpers/HazelcastTestUtils.java

```diff
@@ -167,16 +167,32 @@ public static HazelcastInstanceImpl getHazelcastInstanceImpl(HazelcastInstance h
         return impl;
     }
 
-    public static long nextKeyOwnedBy(long key, HazelcastInstance instance) {
-        final Member localMember = instance.getCluster().getLocalMember();
-        final PartitionService partitionService = instance.getPartitionService();
-        for (; ; ) {
+    /**
+     * Returns the next {@code long} key owned by the given Hazelcast instance.
+     * <p/>
+     * If no key is found after 100 retries a {@link RuntimeException} is thrown.
+     *
+     * @param instance  Hazelcast instance to search next key for
+     * @param lastKey   last key to start search from
+     * @param increment increment to add if next key did not belong to given instance
+     * @return next key owned by given Hazelcast instance
+     * @throws RuntimeException if no key could be found after 100 retries
+     */
+    public static long nextKeyOwnedBy(HazelcastInstance instance, long lastKey, int increment) {
```

Comment:
Hmm, if we don't use an `increment > 1` we cannot reasonably use more than one worker. Otherwise all will share the same key, since the method has no synchronization between the threads (which is good for performance).

If you just want to use multiple threads to increase load on the cluster, we should make that more obvious. Otherwise the usage of this method looks like that all workers are filling the cluster in parallel with different keys. But in fact they don't.